### ISSUE TITLE
Show all error messages in a workspace

### ISSUE THREAD

**Issue by cart** [lapce/lapce#692]
Lapce currently only shows error messages for Rust files that have been opened (and saved). I often open a codebase that already has errors in it. For example: I update rust a dependency, which breaks things in a number of different crates.

When I open a workspace, I want to see the results of cargo check for the entire workspace. This way, if there are build issues I know immediately where to look. And if there are no issues, I can trust the codebase is in a "good state".

With the current "only check open/saved files" approach, I can't use Lapce to determine the state of a codebase and I need to manually run `cargo check` from the command line every time i want to see if there are issues "globally".

This is pretty much the last "big" thing keeping me from adopting Lapce full time.

**Comment by plasticartsshow** [comment]
Would it not be good to have this be an "optional" feature? When using VS Code, for example, checking the whole workspace takes longer, and sometimes seeing all of the information is not helpful when attempting to fix things incrementally.

**Comment by cart** [comment]
> Would it not be good to have this be an "optional" feature? When using VS Code, for example, checking the whole workspace takes longer, and sometimes seeing all of the information is not helpful when attempting to fix things incrementally.

Optional does seem reasonable, but I would personally advocate for "show all workspace errors" as the default. I'm guessing most developers won't want to poke around a workspace (or manually run cargo commands) to find what is broken. I personally wouldn't rely on an error panel that I can't trust to show me all errors in the project I am working on.

**Comment by cart** [comment]
>  I personally wouldn't rely on an error panel that I can't trust to show me all errors in the project I am working on.

Small correction here: I personally wouldn't rely on an error panel that won't show me _some_ error message when there is something broken in my project. It doesn't need to show every error for every file, but it does need to point me to _something_ that is broken, whenever there is something broken. Ex: `cargo build` for a project doesn't always show every error in a codebase. It will generally show me the first error at the earliest possible point in the dependency tree. Once I fix that, I can run `cargo build` again, and either new errors show up (indicating that there are still more issues to fix) or nothing shows up (indicating that everything is fine). Imo the lapce error panel should use the same workflow.

**Comment by dzhou121** [comment]
Isn't the "Problem Panel" show all the errors and warnings of a workspace for you?

**Comment by MinusGix** [comment]
On startup, RA seems to pick up on diagnostics and report them. However, once it starts doing some of the initializing they disappear. So, on startup, you don't see errors in the workspace. However I do see the diagnostics in the workspace after I save a file.  
We might need to see whether we're incorrectly clearing out the diagnostics somewhere, or if we need to tell RA that it should check for errors after initialization?

**Comment by dzhou121** [comment]
Just had a look at the diagnostics code. They can only be "cleared" if RA sends empty diagnostics for a file. 

Also, I've never experienced this issue. The warnings and errors are reported by RA on startup and they don't disappear. Could it be a platform specific issue?

**Comment by MinusGix** [comment]
Hm, testing a bit more and it seems to occur most often when I have an error in another file (ex: I tested with `let x: u64 = 5u32;`) and closed it. Then when I launch Lapce again (so the file isn't open when Lapce is opened), the error appears in the problems listing then disappears after a short period of time. If the file was open in Lapce on startup, it seems to properly display the error.  
I'd be surprised if this is a platform specific issue. If you still don't experience it then I'd expect some weirdness with Rust versions, but I also don't expect that to be issue.

**Comment by cart** [comment]
> Hm, testing a bit more and it seems to occur most often when I have an error in another file (ex: I tested with let x: u64 = 5u32;) and closed the file and then Lapce.

Yup this is how I tested this. I can also reproduce if i introduce an error in a different editor (vscode) and then open the same project in Lapce (and ensure that the file with the error isn't already opened in lapce).
